src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientConfiguration.java

```diff
@@ -61,4 +65,22 @@
         description = "The maximum number of concurrent HTTP connections the connector can make",
         type = AttributeType.INTEGER)
     int maxHttpConnections() default MAX_HTTP_CONNECTIONS_DEFAULT;
+
+    @AttributeDefinition(
+        name = "Http connection timeout",
+        description = "The timeout in milliseconds for the http request",
+        type = AttributeType.INTEGER)
+    int conn_timeout() default DEFAULT_CONN_TIMEOUT;
+
+    @AttributeDefinition(
+        name = "Http socket timeout",
+        description = "The timeout in milliseconds for the socket",
```

Comment:
Same: `The socket timeout in milliseconds, which is the timeout for waiting for data or, put differently, a maximum period inactivity between two consecutive data packets. A timeout value of zero is interpreted as an infinite timeout.`

Reply:
Done